packages/cloudflare/src/index.ts

```diff
@@ -177,7 +184,13 @@ export default function createIntegration(args?: Options): AstroIntegration {
 					});
 				}
 			},
-			'astro:build:setup': ({ vite, target }) => {
+			'astro:build:setup': async ({ vite, target }) => {
+				if (args?.platformProxy?.enabled === true) {
+					const platformProxy = await createPlatformProxy(args.platformProxy);
+
+					setProcessEnv(_config, platformProxy.env);
+				}
+
```

Comment:
@florian-lefebvre why do we need the platformProxy inside of `astro:build:setup`?

Reply:
Yeah I took the easy way. Basically during build, it seems that `vars` in the wrangler config are not loaded so it fails although it shouldn't. We need to override `process.env` at some point in the build but yeah, idk if there's a better way to get the env there

Reply:
@dario-piotrowicz any idea? I don't think it would be super safe to use `platformProxy` here, since you wouldn't get access to your remote configured env vars & bindings 🤔

Additionally we probably only would want to get env vars and not all bindings, but I don't think that is added yet.. I saw that enviroments support is added, correct?

Reply:
> any idea? I don't think it would be super safe to use platformProxy here, since you wouldn't get access to your remote configured env vars & bindings 🤔

sorry just so that I have the full context, what is this for? does Astro inline the env variables or something like that? (or is this about ssg?)

You're right that `platformProxy` contains local simulations so it's not really safe to use during build time, but this is true for ssg and actual bindings usage (e.g. if the code needs to be used for ssg and it can rely on a KV store), if you're only talking about env variables it can be safe-ish to use if you also enforce the use of `pages_build_output_dir` (since the local configuration will become the source of truth in that case)

> Additionally we probably only would want to get env vars and not all bindings, but I don't think that is added yet..

You're referring to https://github.com/cloudflare/workers-sdk/issues/4792 right?
If that's the case yeah it is not yet implemented... I did attempt implementing that a couple of times but I received pushbacks manly for two reasons, the fact that Astro is the only one  requesting it and the fact that we'll need to support all binding types, even future ones, and the team was concerned about the maintainability costs

I'm totally happy to revisit this, especially it if is becoming a blocker for you

> I saw that enviroments support is added, correct?

Yes, in `getPlatformProxy` you can set the environment to use (https://developers.cloudflare.com/workers/wrangler/api/#parameters-1) 🙂👍 
(but it is up to the Astro plugin to set that up)